indigo_api/migrations/0009_migrate_attachment_eids.py

```diff
@@ -0,0 +1,54 @@
+# Generated by Django 2.2.20 on 2021-10-29 11:35
+
+from django.db import migrations
+from indigo_api.data_migrations import CorrectAttachmentEids
+
+
+def forwards(apps, schema_editor):
+    """ Migrate attachment eIds, and change annotations to no longer use the attachment scope prefix.
+    """
+    from indigo_api.models import Document
+    from indigo_api.models import Annotation
+
+    db_alias = schema_editor.connection.alias
+    migration = CorrectAttachmentEids()
+
+    for doc in Document.objects.using(db_alias):
+        if migration.migrate_document(doc):
+            doc.save()
+
+            # update annotations
+            for old, new in migration.eid_mappings.items():
+                # update sec_1 -> att_1__sec_1
+                Annotation.objects.using(db_alias)\
+                    .filter(document_id=doc.id, anchor_id=old)\
+                    .update(anchor_id=new)
+
+                # update prefixed anchors: att_1/sec_1 -> att_1__sec_1
+                prefix = new.split('__', 1)[0] + '/'
+                assert(prefix.startswith('att_'))
+                old = prefix + old
+                new = prefix + new
```

Comment:
prefix ends in /, and we don't want that in `new`, right?

Reply:
I've updated the comment. We preserve the `att_1/att_1__foo` here, and then strip them everywhere with SQL in the next step.

The SQL ensures that we update all annotations, not just those for re-written documents.